gateway/ln-gateway/src/lib.rs

```diff
@@ -194,22 +237,58 @@ impl LnGateway {
         );
 
         let federation_id = FederationId(client.config().client_config.federation_name);
-        self.actors.insert(federation_id, actor.clone());
+        if let Ok(mut actors) = self.actors.lock() {
+            actors.insert(federation_id, actor.clone());
+        }
         Ok(actor)
     }
 
-    fn select_actor(&self, federation_id: FederationId) -> Result<Arc<GatewayActor>> {
-        self.actors
-            .get(&federation_id)
-            .cloned()
-            .ok_or(LnGatewayError::UnknownFederation)
+    // Webserver handler for requests to register a federation
+    async fn handle_register_federation(&self, payload: RegisterFedPayload) -> Result<()> {
+        let connect: WsFederationConnect =
+            serde_json::from_str(&payload.connect).expect("Invalid federation connect info");
+        let api = WsFederationApi::new(connect.members);
+
+        let client_cfg: ClientConfig = api
+            .request(
+                "/config",
+                (),
+                CurrentConsensus::new(api.peers().one_honest()),
+            )
+            .await
+            .expect("Failed to get client config");
+
+        let mut rng = thread_rng();
+        let ctx = secp256k1::Secp256k1::new();
+        let kp_fed = KeyPair::new(&ctx, &mut rng);
+
+        let gw_client_cfg = GatewayClientConfig {
+            client_config: client_cfg,
+            redeem_key: kp_fed,
+            timelock_delta: 10,
+            node_pub_key: self.pub_key,
+            api: Url::parse(format!("http://{}", self.bind_addr).as_str())
+                .expect("Could not parse URL to generate GatewayClientConfig API endpoint"),
+        };
+
+        let client = self.client_builder.build(gw_client_cfg)?;
```

Comment:
Am I missing something or should we save the config somewhere around here so we remember the registered federations after a restart?

Reply:
Ah yes. We should use `builder.save_config()` here, on successful build of the config. The story is incomplete because we need to use builder to rehydrate previous registered federation, from any saved configs present.
We can comprehensively follow up on that bit as part of #817